src/libs/mapChildren.ts

```diff
@@ -0,0 +1,31 @@
+import {Children, ReactNode} from 'react';
+
+/**
+ * A wrapper over `Children.map` which preserves the `ReactNode` structure.
+ *
+ * @param children - a `ReactNode` children structure (i.e. either a single React node or an iterable of React nodes)
+ * @param fn - a mapping function
+ *
+ * @returns - a single mapped React node if `children` was a single node, or an iterable of mapped React nodes if
+ *            `children` was an iterable of React nodes
+ */
+function mapChildren(children: ReactNode, fn: (child: ReactNode) => ReactNode): ReactNode {
```

Comment:
I'm not sure how it would behave if `fn` mapped `child` to an iterable of nodes 🙁

Reply:
I think `mappedChildren` would be a multi-element array in such case

Reply:
Could you clarify what case exactly do you mean? In such scenario, are we afraid of types mismatch?

Reply:
I think it could break the assumption about `mappedChildren` being a one-element-array. I tried to prevent that in my original snippet with the `SingleReactNode` type.

I haven't tested this; I conclude from `Children.map` docs

Reply:
I don't know who thought that a node-or-maybe-many-nodes is a cool data structure. I probably wouldn't get along with them.

Reply:
Yeah according to types, the return value might be an iterator. I've looked through the [source code of `Children.map`](https://github.com/facebook/react/blob/1d5667a1273386f84e416059af7b6aba069e068e/packages/react/src/ReactChildren.js#L77) (and internally used `mapIntoArray`) and if I am not mistake, it looks like iterators are eventually packed into an array, so we don't need to worry about it. 

In that case, we would need to check whether return value of `Children.map` is an array and consist only single item - same goes for its children recursively. 

Or not to reinvent the wheel (in this case a loop :D), I would go back to previous inline solution just for the `applyStrikeThrough`: 

```jsx
// If we only have one child, we want to return it as is, not as an array
if (Array.isArray(strikedThroughChildren) && strikedThroughChildren.length === 1) {
  return strikedThroughChildren[0];
}
```

This way we avoid iterating through mapped children recursively to extract them - we are doing it as the mapping progesses.

Additionally, `Children.map` is kind of a complex helper and patching it like that might not be what we want in the long run.

Let me know what you think 👍

Reply:
I 💯 agree, let's do that!

Reply:
@cubuspl42 here is a commit that adds such solution: https://github.com/Expensify/App/pull/33257/commits/ba7319c6ff1b8693197eae511c7054e596111e62